molgenis-data/src/test/java/org/molgenis/data/meta/AbstractSystemEntityTest.java

```diff
@@ -0,0 +1,243 @@
+package org.molgenis.data.meta;
+
+import static java.lang.String.format;
+import static org.molgenis.data.meta.SystemEntityTestUtils.getReturnTypes;
+import static org.molgenis.data.meta.SystemEntityTestUtils.getTestValue;
+import static org.molgenis.data.system.model.RootSystemPackage.PACKAGE_SYSTEM;
+import static org.testng.Assert.fail;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.ArrayUtils;
+import org.molgenis.data.AbstractMolgenisSpringTest;
+import org.molgenis.data.Entity;
+import org.molgenis.data.EntityFactory;
+import org.molgenis.data.meta.model.Attribute;
+import org.molgenis.data.util.EntityTypeUtils;
+import org.molgenis.util.Pair;
+
+public abstract class AbstractSystemEntityTest extends AbstractMolgenisSpringTest {
```

Comment:
There are multiple warnings in this file (access, unused, etc.). Please fix or suppress